### ISSUE TITLE
Streaming

### ISSUE THREAD

**Issue by sachinkum** [espnet/espnet#982]
I've tried streaming with Transformers, BRNNs and RNN. None of them predict the same thing that is predicted with the complete audio. I've tried 100ms and 200ms chunks as well as (t-1, t), (t, t+1) and (t-1, t+1) chunks. Is there any way to actually use this for streaming yet?

**Comment by sw005320** [comment]
We have been working on the streaming process but we could not completely finish our implementation. Could you let us know which techniques you're working on more specifically? We may figure it out.

**Comment by sachinkum** [comment]
While using transformer (e2e-transformer class), execution results in attribute errors if using window or segment streaming. 
With BRNNs and RNNs, window streaming results in error **AttributeError: 'WindowStreamingE2E' object has no attribute 'retrieve_recognition'**.

With segment streaming, everything fininshes without any errors, but rsults in a blank decode output. 

So while using the plain **model.recognise()**, I've tried to pass small chunks of a single audio file to simulate streaming, but the output that I get using this method is different than what I get if I pass the whole audio. Similarly I've tried passing different combinations of chunks but none give the same output. So if I were to use this method, should I train the network on only **chunk lenght** audio? But that would defeat the whole purpose of E2E. If not, is there a way that the attention layer could keep previous attention values to recompute them on new chunk? 
I'm trying to implement a streaming system so I can't have too much latency.
